### ISSUE TITLE
Unit test:  temporal search returns correct content

### ISSUE THREAD

**Issue by agstephens** [cedadev/esgf-stac-client#8]
## Overview

Temporal search (or search over time) should match appropriate responses at both the `item` and `asset` level. We need to check that providing a time window will limit the responses returned by search. 

We can think of the time window as being:
- `../..` or unspecified: match all times
- `<start>/..`: start bound only - match all times after `<start>`
- `../<end>`: end bound only - match all times up to `<end>`
- `<start>/<end>`: match all times within period specified

You can also specify a fixed time point:
- `<datetime>` - just a single value

**NOTE: bounds should be inclusive - if a timestep matches exactly the `<start>` or `<end>` then that record should be included.**

### Item tests

In order to get a small number of results, use a time period in the future:

```
result = client.search(datetime="3500-01-01/3600-12-30")
result.matched()
32
```

You can test for compliance by looking at:

```
item = next(result.items())
[(prop, item.properties[prop]) for prop in item.properties.keys() if "date" in prop]

[('start_datetime', '3180-01-01T00:00:00.000Z'),
 ('end_datetime', '3580-12-01T00:00:00.000Z')]
```

Positive results:
- [x] Test without `datetime` argument at all
- [x] Test with single datetime
- [x] Test lower-bounded
- [x] Test upper-bounded
- [x] Test full time window

In each case, loop through all the records returned were all in the expected time selection.

Now, in order to test a negative response, specify a single record with:

```
client.search(instance_id="CMIP6.CMIP.NASA-GISS.GISS-E2-1-H.piControl.r1i1p1f1.Omon.zostoga.gn.v20190410")
```

Check the valid time window within the record, and now check that the various tests return no matches when you specify time windows that do NOT overlap at all. Combine your `datetime` argument with `instance_id` as above).

Negative results (should return no records):
- [ ] Test with single datetime
- [ ] Test lower-bounded
- [ ] Test upper-bounded
- [ ] Test full time window

### Asset tests

Apply the same tests as above but with the following modifications:

1. Create the search with `client.asset_search(...)`
2. Within a single `item` record, that has more than one `asset`, call `item.get_assets(datetime=<datetime>)` - and check that the asset date/time values match the time selections.

**Comment by agstephens** [comment]
Hi @Adik8688: here is the first of a more detailed unit test. Sorry it took me so long!

**Comment by Adik8688** [comment]
For last 4 tests cases I was passing **instance_id** and **datetime** to **client_search()**
What's an equivalent of **instance_id** for **item.asset_search()**?

**Comment by agstephens** [comment]
Some notes on Asset Search:

```
# Use either:
asearch = client.asset_search()
# or 
asearch2 = client.search(doctype="files") # or doctype="assets"

assert type(asearch) == type(asearch2)
print(type(asearch))
# Use list comprehension to grab N assets - in this case 10
assets_10 = [next(asearch.get_assets()) for i in range(10)]

# # View the last asset
asset = assets_10[-1]
asset
```

**Comment by Adik8688** [comment]
Still, I'm not sure sure how it applies to the last 4 ticks
For item I was using certain combination of instance_id and datetime to make sure I won't get back any result, as I expect
So analogically I should combine datetime and some kind of asset ID to reach the same logic for asset_search()

**Comment by agstephens** [comment]
Good point @Adik8688, I'll try to get something useful and put it in the ticket.

**Comment by agstephens** [comment]
@Adik8688:  I think this is broken on the server-side so implement the tests as follows:
- `...asset_search(item=<item>, datetime=<datetime>)` - instance of item
- `...asset_search(item=[<item>], datetime=<datetime>)` - instance of item(s) in a list
- `...asset_search(item=<item_id>, datetime=<datetime>)` - the `instance_id` of a known item
- `...asset_search(item=[<item_id>], datetime=<datetime>)` - a list of the `instance_id` of a known item

Let's let them fail.
